cmd/containerd/config.go

```diff
@@ -1,11 +1,32 @@
 package main
 
 import (
+	"bytes"
+	"io"
 	"os"
 
+	"github.com/BurntSushi/toml"
+	"github.com/containerd/containerd/server"
 	"github.com/urfave/cli"
 )
 
+// Config is a wrapper of server config for printing out.
+type Config struct {
+	*server.Config
+	// Plugins overrides `Plugins map[string]toml.Primitive` in server config.
+	Plugins map[string]interface{} `toml:"plugins"`
+}
+
+// WriteTo marshals the config to the provided writer
+func (c *Config) WriteTo(w io.Writer) (int64, error) {
+	buf := bytes.NewBuffer(nil)
+	e := toml.NewEncoder(buf)
```

Comment:
Why use a buffer here and not just `w`?

Reply:
Copied from original code. Will remove it then.

Reply:
Done.